### PR TITLE
Scripts for the AG sentinel camp QB wall

### PR DESCRIPTION
**Pull request by aronwk-aaron**
Two scripts used in tandem for spawning the mobs outside the QB wall in the sentinel outpost area.

Tested that the enemies aggro to the gate on spawn then aggro to the player that completed the QB
Tested that if they are still alive and the QB is rebuilt, it re-aggros the enemies and only spawns new ones if the have been killed

### CODE REVIEW

File: dScripts/QbSpawner.cpp [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+			// enemy:FollowTarget{targetID = gateObj}
+			auto* movementAIComponent = enemy->GetComponent<MovementAIComponent>();
+			if (movementAIComponent) movementAIComponent->SetDestination(gate->GetPosition());
+			baseCombatAIComponent->Taunt(gateObjID, 1000);
+		}
+	}
+
+	auto playerObjID = self->GetVar<LWOOBJID>(u"player");
+	if (playerObjID) {
+		baseCombatAIComponent->Taunt(playerObjID, 1000);
```
Comment (EmosewaMC) [inline dScripts/QbSpawner.cpp:140]: The original script values this as 100

File: dScripts/QbSpawner.cpp [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+	Game::logger->Log("QbSpawner", "child removed");
+	auto mobTable = self->GetVar<std::vector<LWOOBJID>>(u"mobTable");
+	auto tableLoc = child->GetVar<int>(u"mobTableLoc");
+
+	mobTable[tableLoc] = LWOOBJID_EMPTY;
+	self->SetVar<std::vector<LWOOBJID>>(u"mobTable", mobTable);
+}
+
+void QbSpawner::AggroTargetObject(Entity* self, Entity* enemy) {
+	Game::logger->Log("QbSpawner", "Aggro");
```
Comment (EmosewaMC) [inline dScripts/QbSpawner.cpp:122]: Note to remove logs and relevant headers

File: dScripts/QbSpawner.cpp [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+
+		auto* gate = EntityManager::Instance()->GetEntity(gateObjID);
+		if (!gate) return;
+
+		auto oPos = gate->GetPosition();
+		auto oDir = gate->GetRotation().GetForwardVector();
+		auto newPos = NiPoint3::ZERO;
+		newPos.x = oPos.x + (oDir.x * spawnDist);
+		newPos.y = oPos.y;
+		newPos.z = oPos.z + (oDir.z * spawnDist);
```
Comment (EmosewaMC) [inline dScripts/QbSpawner.cpp:53]: This should be contatenated to `NiPoint3 newPos(oPos.x + (oDir.x * spawnDist), oPos.y, oPos.z + (oDir.z * spawnDist));`

File: dScripts/QbSpawner.cpp [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+
+		auto newRot = NiQuaternion::LookAt(newPos, oPos);
+
```
Comment (EmosewaMC) [inline dScripts/QbSpawner.cpp:56]: Extra blank line

File: dScripts/QbSpawner.cpp [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+
+		for (int i = 0; i < mobTable.size(); i++) {
+			Game::logger->Log("QbSpawner", "spawn new %i", mobTemplate);
+			int posOffset = -10;
```
Comment (EmosewaMC) [inline dScripts/QbSpawner.cpp:60]: Can be made constant

File: dScripts/QbSpawner.cpp [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+				info.settings = {
+					new LDFData<bool>(u"no_timed_spawn", true),
+					new LDFData<float>(u"aggroRadius", 70),
+					new LDFData<float>(u"softtetherRadius", 80),
+					new LDFData<float>(u"tetherRadius", 90),
+					new LDFData<float>(u"wanderRadius", 5),
+					new LDFData<int>(u"mobTableLoc", i)
+				};
+
+				auto* entity = EntityManager::Instance()->CreateEntity(info, nullptr, self);
```
Comment (EmosewaMC) [inline dScripts/QbSpawner.cpp:79]: This should be called `child`, `entity` is ambiguous

File: dScripts/QbSpawner.cpp [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+
+	AggroTargetObject(self, child);
+
+	const auto selfID = self->GetObjectID();
+
+	child->AddDieCallback([this, selfID, child]() {
+		auto* self = EntityManager::Instance()->GetEntity(selfID);
+		OnChildRemoved(self, child);
+		}
+	);
```
Comment (EmosewaMC) [inline dScripts/QbSpawner.cpp:109]: Extra new line here

File: dScripts/QbSpawner.cpp [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+
+	auto gateObjID = self->GetVar<LWOOBJID>(u"gateObj");
+	if (gateObjID) {
+		auto* gate = EntityManager::Instance()->GetEntity(gateObjID);
+		if (gate) {
+			// Not sure if this is the same as, but it should work?
+			// enemy:FollowTarget{targetID = gateObj}
+			auto* movementAIComponent = enemy->GetComponent<MovementAIComponent>();
+			if (movementAIComponent) movementAIComponent->SetDestination(gate->GetPosition());
+			baseCombatAIComponent->Taunt(gateObjID, 1000);
```
Comment (EmosewaMC) [inline dScripts/QbSpawner.cpp:134]: Yes these can be interchanged.

File: dScripts/QbSpawner.cpp (hunk 5 above)
Comment (aronwk-aaron) [reply dScripts/QbSpawner.cpp:60]: No cause it changes throughout the loop

Comment (EmosewaMC) [reply dScripts/QbSpawner.cpp:60]: Ah I see, its used to cycle positions

File: dScripts/QbSpawner.cpp (hunk 1 above)
Comment (aronwk-aaron) [reply dScripts/QbSpawner.cpp:140]: copy/pasta error, ty

**Comment by aronwk-aaron** [comment]
ty for the feedback.
current issue is that the enemies aren't... correct(? they aren't visible after spawning) so that is why I still have logs and this is a WIP PR

**Review by EmosewaMC — APPROVED** [review]

**Pull request merged**